### DISCUSSION TITLE
Delay Pre-rendering and SEO

### DISCUSSION THREAD

**Discussion by mousedown** [vercel/next.js#24955] [Help] [answered]
Looking for some help understanding this please. I have a site with pre-renderd dynamic pages (Incremental Static Regeneration), These work great, except for the very first request. Instead of the SSR page I expected, this first returns the skeleton page whilst the page is rendered, the quickly updates. The second request serves the cached page quickly. The problem here, is that the 1st page in most cases accessed by google first , and the next-seo hasn't been generated so the canonical name isn't set. 

Is it possible to bypass this quirk and ensure 1st page load is delayed whilst pre-rendering or would it be best to writer a service to ping the pages forcing rendering so that google will always load a page from the edge? 

age: 0
cache-control: public, max-age=0, must-revalidate
content-disposition: inline
content-encoding: br
content-type: text/html; charset=utf-8
date: Mon, 10 May 2021 16:11:01 GMT
server: Vercel
strict-transport-security: max-age=63072000
x-matched-path: /[type]/[slug]
x-vercel-cache: PRERENDER

**Comment by timneutkens** [comment, marked as answer]
You're most likely using `fallback: true` but you're looking for `fallback: 'blocking'`: https://nextjs.org/docs/basic-features/data-fetching#fallback-blocking 👍

**Reply by mousedown** [reply]
Amazing! I didn't know about this feature. 
Thanks @timneutkens !
